.github/workflows/typing.yml

```diff
@@ -0,0 +1,46 @@
+name: typing
```

Comment:
Since we will have several packages inside the repo, currently the precommit is a global file in the repo root, and we might need specific config for each package. Will defer for a future PR. Thanks!